mautic/mautic
# Add Fresh Consulting WCAG 2.0 Linter

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No 
| New feature/enhancement? (use the a.x branch)      | Yes 
| Deprecations?                          | No
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | No 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
The PR adds [FreshConsulting WCAG 2.0 Linter](https://github.com/FreshConsulting/FreshConsulting-WCAG-Linter). Using this linter we can make sure that the current code is compliant with the rules of [FreshConsulting/FreshConsulting-WCAG-Linter](https://github.com/FreshConsulting/FreshConsulting-WCAG-Linter): (A PHP_CodeSniffer custom rule (sniff) to detect obvious WCAG 2.0 accessibility violations) and newly committed code will be compliant with the rules.

All linter errors have been fixed.



#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Run composer install
3. Add the repository path to the PHP_CodeSniffer configuration: 
```
phpcs --config-set installed_paths /var/www/html/vendor/fresh-consulting/fresh-consulting-wcag-linter
```
4. Run either
```
phpcs /path/to/code
```
   or
```
phpcs --sniffs=FreshConsulting.WCAG20.Violations /path/to/code
```
5. No linter errors



<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10818"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
